service/deployer/deployer.go

```diff
@@ -50,12 +59,21 @@ func New(config Config) (Deployer, error) {
 		return nil, microerror.MaskAnyf(invalidConfigError, "logger must not be empty")
 	}
 
+	// Settings.
+	if config.Flag == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "flag must not be empty")
+	}
+	if config.Viper == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "viper must not be empty")
+	}
```

Comment:
I'm checking clients where they're needed - e.g: the slack notifier checks if the slack client is nil. Idea here is to be more flexible in the future. If the user uses a different notifier, they shouldn't need to configure a slack client.